spras/evaluation.py

```diff
@@ -82,120 +80,159 @@ def load_files_from_dict(self, gold_standard_dict: Dict):
+                # Dropping last elements because scikit-learn adds (1, 0) to precision/recall for plotting, not tied to real thresholds
+                # https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_curve.html#sklearn.metrics.precision_recall_curve:~:text=Returns%3A-,precision,predictions%20with%20score%20%3E%3D%20thresholds%5Bi%5D%20and%20the%20last%20element%20is%200.,-thresholds
+                prc_data = {
                     'Threshold': thresholds,
                     'Precision': precision[:-1],
                     'Recall': recall[:-1],
-                    'Average_Precison': avg_precision,
-                    'Baseline_Precision': baseline_precision
                 }
 
-                if label != 'ensemble':
-                    data['Algorithm'] = [label] * len(thresholds)
-                    cols = ['Algorithm'] + [col for col in data if col != 'Algorithm']
-                else:
-                    cols = list(data.keys())
-
-                df = pd.DataFrame(data)[cols]
-                dfs.append(df)
-
-            else:
-                plt.plot([], [], color=color_palette[label], marker='o', label=f'{label} PR curve - Empty Ensemble')
-
-                data = {
-                    'Threshold': ["None"],
-                    'Precision': ["None"],
-                    'Recall': ["None"],
-                    'Average_Precison': ["None"],
-                    'Baseline_Precision': ["None"]
+                metric_data = {
+                    'Average_Precision': [avg_precision],
+                    'AUC': [auc_score]
                 }
 
-                if label != 'ensemble':
-                    data['Algorithm'] = [label]
-                    cols = ['Algorithm'] + [col for col in data if col != 'Algorithm']
-                else:
-                    cols = list(data.keys())
+                ensemble_source = label.capitalize() if label != 'ensemble' else "Aggregated"
+                prc_data = {'Ensemble_Source': [ensemble_source] * len(thresholds), **prc_data}
+                metric_data = {'Ensemble_Source': [ensemble_source], **metric_data}
 
-                df = pd.DataFrame(data)[cols]
-                dfs.append(df)
+                prc_df = pd.DataFrame.from_dict(prc_data)
+                prc_dfs.append(prc_df)
+                metric_df = pd.DataFrame.from_dict(metric_data)
+                metric_dfs.append(metric_df)
 
+            else:
+                raise ValueError(
+                    "Cannot compute PRC curve: the ensemble network is empty."
+                    f"This should not happen unless the input network for pathway reconstruction is empty"
+                )
+
+        baseline_precision = np.sum(y_true) / len(y_true) # same baseline value for all algorithms since all network nodes are included in the ensemble
+        plt.axhline(y=baseline_precision, color="black", linestyle='--', label=f'Baseline_Precision: {baseline_precision:.4f}')
```

Comment:
This is formatted differently. It doesn't have the "AP:" prefix or parentheses. We can drop "_Precision" because AP is the metric.